src/pybind/mgr/cephadm/module.py

```diff
@@ -2759,7 +2759,15 @@ def add_mgr(self, spec):
         return self._add_daemon('mgr', spec, self._create_mgr)
 
     def _apply(self, spec: ServiceSpec) -> str:
-        if spec.placement.is_empty():
+
+        if spec.service_type == 'host':
+            completion = self.add_host(spec)
```

Comment:
This should fail badly in mypy. either we change the type hierarchy (i.e. move `HostSpec` and `ServiceSpec` into a common type hierarchy) or we differentiate between hosts and `ServiceSpec` somewhere else.